devito/passes/iet/definitions.py

```diff
@@ -397,7 +398,9 @@ def place_casts(self, iet, **kwargs):
         # Some objects don't distinguish their _C_symbol because they are known,
         # by construction, not to require it, thus making the generated code
         # cleaner. These objects don't need a cast
-        bases = [i for i in bases if i.name != i.function._C_name]
+        bases = [
+            i for i in bases
+            if i.name != i.function._C_name and not isinstance(i.function, PETScArray)]
```

Comment:
I do not know how to avoid using PETScArray explicitly here

Reply:
remind me what's the line that you want to avoid here?